### PR TITLE
refactor: Break Database query tests into their own files + add basic SQL tests

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
I have been collecting tests for queries into their own modules, in preparation for this moment -- splitting them into their own files!

I want to split these tests into their own modules because this way of testing queries is the basic pattern I am planning for all queries going forward.

# Changes
Splits modules out of `db.rs` and into their own files

It also adds some very basic SQL tests -- this is already covered in the end to end test but now we have a skeleton / pattern for doing it without compiling / firing up that behemoth. This was previously done only in the mutable buffer, but I deleted it in https://github.com/influxdata/influxdb_iox/pull/730

I imagine at some point these tests might want to be their own crate too.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: server/src/db.rs [hunk 1]
```diff
@@ -587,339 +573,3 @@ mod tests {
         chunk_ids
     }
 }
-
-#[cfg(test)]
```
Comment (alamb) [inline server/src/db.rs:591]: This PR just moves all this code into new files.

File: server/src/query_tests/sql.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
+//! basic tests of SQL query planning DataFusion contains much more
+//! extensive test coverage, this module is meant to show we have
+//! wired all the pieces together (as well as ensure any particularly
+//! important SQL does not regress)
+
+use super::scenarios::*;
```
Comment (alamb) [inline server/src/query_tests/sql.rs:6]: These are some new tests -- they effectively test sql with data loaded into the mutable and read buffer.

**Review by pauldix — APPROVED** [review]
Looks like a good separation. I'd love to get to the point where we can do this kind of testing via the APIs exclusively and specify the tests in config/setup files that get run, but maybe that's overthinking it.

**Comment by alamb** [comment]
@pauldix  I think driving these queries via the APIs would not be hard (when the APIs exist) -- they don't rely on any of the internal implementation details.

**Comment by alamb** [comment]
@pauldix  I also like the idea of running the tests from external files rather than hard coded expected values

**Pull request merged**